contribute/design/1.x/cstor/20200511-cstor-backupandrestore.md

```diff
@@ -0,0 +1,323 @@
+
+	// BKPCStorStatusPending , backup is pending.
+	BKPCStorStatusPending CStorBackupStatus = "Pending"
+
+	// BKPCStorStatusInProgress , backup is in progress.
+	BKPCStorStatusInProgress CStorBackupStatus = "InProgress"
+
+	// BKPCStorStatusInvalid , backup operation is invalid.
+	BKPCStorStatusInvalid CStorBackupStatus = "Invalid"
+)
+```
+
+### CStorRestore Schema
+Following is the existing CStorRestore schema in go struct:
+```go
+// CStorRestore describes a cstor restore resource created as a custom resource
+type CStorRestore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"` // set name to restore name + volume name + something like csp tag
+	Spec              CStorRestoreSpec            `json:"spec"`
+	Status            CStorRestoreStatus          `json:"status"`
+}
+
+// CStorRestoreSpec is the spec for a CStorRestore resource
+type CStorRestoreSpec struct {
+	RestoreName   string            `json:"restoreName"` // set restore name
+	VolumeName    string            `json:"volumeName"`
+    // RestoreSrc can be ip:port in case of restore from remote or volumeName in case of local restore
+	RestoreSrc    string            `json:"restoreSrc"`
+	MaxRetryCount int               `json:"maxretrycount"`
+	RetryCount    int               `json:"retrycount"`
+	StorageClass  string            `json:"storageClass,omitempty"`
+	Size          resource.Quantity `json:"size,omitempty"`
+    // Local will be helpful to identify whether restore is from local (or) backup/snapshot
+	Local         bool              `json:"localRestore,omitempty"`
+}
+
+// CStorRestoreStatus is to hold result of action.
+type CStorRestoreStatus string
+
+// Status written onto CStrorRestore object.
+const (
+	// RSTCStorStatusEmpty ensures the create operation is to be done, if import fails.
+	RSTCStorStatusEmpty CStorRestoreStatus = ""
+
+	// RSTCStorStatusDone , restore operation is completed.
+	RSTCStorStatusDone CStorRestoreStatus = "Done"
+
+	// RSTCStorStatusFailed , restore operation is failed.
+	RSTCStorStatusFailed CStorRestoreStatus = "Failed"
+
+	// RSTCStorStatusInit , restore operation is initialized.
+	RSTCStorStatusInit CStorRestoreStatus = "Init"
+
+	// RSTCStorStatusPending , restore operation is pending.
+	RSTCStorStatusPending CStorRestoreStatus = "Pending"
+
+	// RSTCStorStatusInProgress , restore operation is in progress.
+	RSTCStorStatusInProgress CStorRestoreStatus = "InProgress"
+
+	// RSTCStorStatusInvalid , restore operation is invalid.
+	RSTCStorStatusInvalid CStorRestoreStatus = "Invalid"
+)
+```
+
+### CStorCompletedBackup Schema
+Following is the existing CStorCompletedBackup schema in go struct:
+```go
+// CStorCompletedBackup describes a cstor completed-backup resource created as custom resource
+type CStorCompletedBackup struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              CStorBackupSpec `json:"spec"`
+}
+```
+
+## Drawbacks
```

Comment:
We can remove `Drawbacks`, `Alternatives` and `Infrastructure Needed`. If needed, can be added later.

Reply:
Removed